### ISSUE TITLE
[Google3 Bug]: State information not announced for AM/PM in Material DateTimePicker

### ISSUE THREAD

**Issue by evantea** [flutter/flutter#173302]
### Help us understand the severity of this issue

- [ ] causing severe production issues e.g. malfunctions or data loss
- [ ] blocking next binary release
- [x] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

1. Turn on Voicover
2. Navigate to the DateTimePicker
3. Select a date
4. Navigate to the AM/PM buttons and select one of them

### Expected results

VoiceOver announces the AM/PM buttons with state information (e.g. selected, PM, button).

### Actual results

VoiceOver announces the AM/PM buttons without state information.

### Code sample
<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(body: Center(child: MyWidget())),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    WidgetsBinding.instance.addPostFrameCallback((timestamp) {
      showTimePicker(
        context: context,
        initialTime: TimeOfDay.now(),
        orientation: Orientation.portrait,
      );
    });

    return Container(height: 300, width: 100);
  }
}
```

</details>

### Screenshots or Video

<img width="550" height="573" alt="Image" src="https://github.com/user-attachments/assets/453799a3-8929-40a4-b25c-70a8a6b4acc0" />

### Flutter Doctor output

n/a

**Comment by evantea** [comment]
This bug affects the "chalk" app

**Comment by evantea** [comment]
The semantics are currently being announced as RadioButton and "checked" and "not-checked", closing for now to verify with our accessibility team that this is enough. I'll re-open with more clear instructions if any action is necessary :)

**Comment by evantea** [comment]
Re-opening! The information provided above is only happening on Android, for iOS it simply says the following no matter the state of the buttons:
- "AM, button"
- "PM, button"

**Comment by tirth-patel-nc** [comment]
@evantea, thanks for the report. We are seeing the same behavior with the latest SDK versions on iOS. My colleague verified it on iOS as well:

```
stable : 3.32.8
master : 3.33.0-1.0.pre-1299
```

| Android | iOS |
| --- | --- |
| checked AM, radio button, double tap to toggle | Same as your report https://github.com/flutter/flutter/issues/173302#issuecomment-3156646733
| not checked PM, radio button, double tap to toggle | Same as your report https://github.com/flutter/flutter/issues/173302#issuecomment-3156646733 |

**Comment by chunhtai** [comment]
looks like the material date picker should use Semantics(selected: true) for iOS platform, should be an easy fix

**Comment by JulioJ11** [comment]
I'd like to attempt on solving this issue and will submit a PR

**Comment by tirth-patel-nc** [comment]
please go ahead, it's all yours. thanks for contributing

**Comment by evantea** [comment]
Hi y'all,
Any update on this issue?
